optuna/storages/_rdb/alembic/versions/v3.0.0.a.py

```diff
@@ -0,0 +1,216 @@
+from optuna.distributions import IntDistribution
+from optuna.distributions import IntLogUniformDistribution
+from optuna.distributions import IntUniformDistribution
+from optuna.distributions import json_to_distribution
+from optuna.distributions import LogUniformDistribution
+from optuna.distributions import UniformDistribution
+from optuna.trial import TrialState
+
+
+# revision identifiers, used by Alembic.
+revision = "v3.0.0.a"
+down_revision = "v2.6.0.a"
+branch_labels = None
+depends_on = None
+
+
+MAX_INDEXED_STRING_LENGTH = 512
+
+
+BaseModel = declarative_base()
+
+
+class StudyModel(BaseModel):
+    __tablename__ = "studies"
+    study_id = Column(Integer, primary_key=True)
+    study_name = Column(String(MAX_INDEXED_STRING_LENGTH), index=True, unique=True, nullable=False)
+
+
+class TrialModel(BaseModel):
+    __tablename__ = "trials"
+    trial_id = Column(Integer, primary_key=True)
+    number = Column(Integer)
+    study_id = Column(Integer, ForeignKey("studies.study_id"))
+    state = Column(Enum(TrialState), nullable=False)
+    datetime_start = Column(DateTime)
+    datetime_complete = Column(DateTime)
+
+
+class TrialParamModel(BaseModel):
+    __tablename__ = "trial_params"
+    __table_args__: Any = (UniqueConstraint("trial_id", "param_name"),)
+    param_id = Column(Integer, primary_key=True)
+    trial_id = Column(Integer, ForeignKey("trials.trial_id"))
+    param_name = Column(String(MAX_INDEXED_STRING_LENGTH))
+    param_value = Column(Float)
+    distribution_json = Column(Text())
+
+
+def migrate_new_distribution(distribution_json: str) -> str:
+    distribution = json_to_distribution(distribution_json)
+    new_distribution: BaseDistribution
+
+    # Float distributions.
+    if isinstance(distribution, UniformDistribution):
+        new_distribution = FloatDistribution(
+            low=distribution.low,
+            high=distribution.high,
+            log=False,
+            step=None,
+        )
+    elif isinstance(distribution, LogUniformDistribution):
+        new_distribution = FloatDistribution(
+            low=distribution.low,
+            high=distribution.high,
+            log=True,
+            step=None,
+        )
+    elif isinstance(distribution, DiscreteUniformDistribution):
+        new_distribution = FloatDistribution(
+            low=distribution.low,
+            high=distribution.high,
+            log=False,
+            step=distribution.q,
+        )
+
+    # Integer distributions.
+    elif isinstance(distribution, IntUniformDistribution):
+        new_distribution = IntDistribution(
+            low=distribution.low,
+            high=distribution.high,
+            log=False,
+            step=distribution.step,
+        )
+    elif isinstance(distribution, IntLogUniformDistribution):
+        new_distribution = IntDistribution(
+            low=distribution.low,
+            high=distribution.high,
+            log=True,
+            step=distribution.step,
+        )
```

Comment:
What happens when `IntLogUniformDistribution.log == True` and `IntLogUniformDistribution.step != 1`?

(@HideakiImamura memtioned that such inconsistent distributions  were not accepted by released versions of Optuna.)

Reply:
@HideakiImamura suggested that we can implicitly convert `step=1`. The minimum value of `step` is `1`, and the corresponding parameter values are supposed to be included by the converted distributions.

Reply:
@himkt suggested that it should be failed due to the invalid distribution arguments. The migration should be stopped, and users will delete the corresponding studies by themselves.

Reply:
TODO: We'd like to check if we can re-run the migration script after removing the studies with invalid trials.

Reply:
I confirmed that we could re-run the migration script if it failed. So, users can re-run the migration after removing studies with invalid trials.